src/puppet.ts

```diff
@@ -378,8 +362,71 @@ export abstract class Puppet extends PuppetEventEmitter {
    *
    *
    */
-  public abstract async start () : Promise<void>
-  public abstract async stop ()  : Promise<void>
+  public async start () : Promise<void> {
+    this.on('heartbeat', this.feedDog)
+    this.watchdog.on('reset', this.dogReset)
+
+    /**
+     * 2. Setup `reset` Event via a 1 second Throttle Queue:
+     */
+    this.resetThrottleQueue = new ThrottleQueue<string>(1000)
```

Comment:
How about let's keep the `resetThrottleQueue` stay in our `constructor` because I believe there's no need to initialize it in start() and destroy it in stop()?

Please correct me if I'm wrong.

Reply:
Is there a way to clear the queue? If so, we can add a clear operation in the `stop()`, then we can keep it in the constructor. Otherwise, I am worried about the queue is not fully emptied and trigger some weird behaviors after a reset.

Reply:
I understand your concern.

I believe that we do not care to clear the queue at all.

Here's the reason: we are using a ThrottleQueue for events. A ThrottleQueue will process the first event and block all following events in a period of time.

Whenever we stop the event emitter at any time point, the `resetThrottleQueue` will not emit any future events anymore.

See: https://github.com/huan/rx-queue#throttlequeue

Reply:
Yes, that makes sense, I will move it to constructor.